src/lib/dev-environment/types.js

```diff
@@ -0,0 +1,54 @@
+
```

Comment:
General comment on this type definitions, which cascade all over the PR -- could we try to have less nullable types? It would save us a lot of `x?.something || ' '`.